src/System.Runtime.Serialization.Formatters/tests/BinaryFormatterTests.cs

```diff
@@ -48,95 +29,31 @@ public void ValidateBasicObjectsRoundtrip(object obj, FormatterAssemblyStyle ass
-                {
-                    bf.Serialize(ms, obj[0]);
-                    string serializedHash = Convert.ToBase64String(ms.ToArray());
-                    serializedHashes.Add(serializedHash);
-                }
-            }
+            string testDataFilePath = GetTestDataFilePath();
+            IEnumerable<object[]> coreTypeRecords = GetCoreTypeRecords();
+            string[] coreTypeBlobs = GetCoreTypeBlobs(coreTypeRecords).ToArray();
 
-            string path = @"D:\corefx\src\System.Runtime.Serialization.Formatters\tests\BinaryFormatterTestData.cs";
-            string[] lines = File.ReadAllLines(path);
-
-            List<string> newLines = new List<string>();
-            int numOfHashes = 0;
-            for (int i = 0; i < lines.Length; i++)
-            {
-                string line = lines[i];
-                if (!line.Trim().StartsWith("yield") || numOfHashes >= serializedHashes.Count)
-                {
-                    newLines.Add(line);
-                    continue;
-                }
-
-                if (PlatformDetection.IsFullFramework)
-                {
-                    line = Regex.Replace(line, ", \"AAEAAAD.+\"(?!,)", ", \"" + serializedHashes[numOfHashes] + "\""); // netfx
-                }
-                else
-                {
-                    line = Regex.Replace(line, "\"AAEAAAD.+\",", "\"" + serializedHashes[numOfHashes] + "\","); // netcoreapp
-                }
-
-                newLines.Add(line);
-                numOfHashes++;
-            }
-
-            Assert.Equal(numOfHashes, serializedHashes.Count);
-
-            File.WriteAllLines(path, newLines);
-        }
-
-
-        private static string SerializeObjectToHash(object original)
-        {
-            BinaryFormatter bf = new BinaryFormatter();
-            using (MemoryStream ms = new MemoryStream())
-            {
-                bf.Serialize(ms, original);
-                return Convert.ToBase64String(ms.ToArray());
-            }
-        }
-
-        private static object DeserializeObjectHash(string base64Str)
-        {
-            var binaryFormatter = new BinaryFormatter();
-            byte[] serializedObj = Convert.FromBase64String(base64Str);
-            using (var serializedStream = new MemoryStream(serializedObj))
-            {
-                return binaryFormatter.Deserialize(serializedStream);
-            }
+            UpdateCoreTypeBlobs(testDataFilePath, coreTypeBlobs);
         }
 
         [Theory]
         [MemberData(nameof(SerializableObjects))]
-        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework)] 
-        public void ValidateTfmHashes(object original, string[] tfmBase64Hashes)
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework)]
+        public void ValidateAgainstBlobs(object obj, string[] blobs)
         {
-            if (tfmBase64Hashes == null || tfmBase64Hashes.Length < 2)
+            if (blobs == null || blobs.Length < 2)
```

Comment:
I wanted to make sure that we have at least a core and netfx blob. In future we e.g. could have a mono blob for some types therefore I prefer < 2 instead of != 2. As you suggested in the other comments I will fine tune the error message.